frontend/src/components/object/ObjectTable.vue

```diff
@@ -64,17 +64,18 @@ const togglePublic = async (objectId: string, isPublic: boolean) => {
   await objectStore.togglePublic(objectId, isPublic);
 };
 
-watch( getObjects, () => {
+watch( getObjects, async () => {
   // Filter object cache to this specific bucket
   const objs: Array<COMSObjectDataSource> = getObjects.value
     .filter( (x: COMSObject) => x.bucketId === props.bucketId ) as COMSObjectDataSource[];
 
-  // Apply metadata
+  // update metadata store
+  await metadataStore.fetchMetadata({objectId: objs.map( (x: COMSObject) => x.id )});
+
```

Comment:
While not a part of your direct code changes, we could take advantage of object reconstruction by using a map instead of performing a forEach and assignment change:
```js
tableData.value = objs.map( (x: COMSObjectDataSource) => {
  x.name = metadataStore.findValue(x.id, 'name');
  return x;
});
```